### PR TITLE
feat: install the cert-manager on k8s

### PR DESCRIPTION
**Pull request by jacobdotcosta**

### CODE REVIEW

File: ansible/playbook/kubernetes/README.adoc [hunk 1]
```diff
@@ -92,6 +92,75 @@ ansible-playbook ansible/playbook/kubernetes/k8s_remove.yml -e vm_name=${VM_NAME
...
+As a result the following resources should be available.
+
+[source]
+----
+$ kubectl -n cert-manager get all
+NAME                                           READY   STATUS    RESTARTS   AGE
+pod/cert-manager-7c869867bf-jlwxj              1/1     Running   0          88s
+pod/cert-manager-cainjector-54c9d9b775-9zhrm   1/1     Running   0          88s
+pod/cert-manager-webhook-7f7469bdb7-m5b5z      1/1     Running   0          88s
+pod/godaddy-webhook-c6b5f74fd-5w7q6            1/1     Running   0          66s
```
Comment (cmoulliard) [inline ansible/playbook/kubernetes/README.adoc:140]: Can you also, part of the description, mention that we install also the Godaddy webhook - https://github.com/snowdrop/k8s-infra/blob/main/ansible/roles/cert_manager/tasks/install.yml#L24 ?

**Comment by jacobdotcosta** [comment]
Improved the docs to describe the OCP certificate renewal process.

File: ansible/playbook/kubernetes/README.adoc (hunk 1 above)
Comment (jacobdotcosta) [reply ansible/playbook/kubernetes/README.adoc:149]: Added to the docs.

File: ansible/roles/k8s_cert_manager/README.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
+== Role: Kubernetes Certificate Manager
+
+Deploy Kubernetes Certificate Manager on kubernetes cluster
+
+=== Description
+
+This role will install the Kubernetes certificate manager on a kubernetes cluster.
+
+It also installes the _Godaddy webhook_.
```
Comment (cmoulliard) [inline ansible/roles/k8s_cert_manager/README.adoc:9]: `It also installes` -> `Remark`: it also installs the Godaddy webhook

File: openshift/certificate_renewal.adoc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+:description: Renew OCP cluster certificate
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+This document describes the process to renew an OCP cluster certificates.
```
Comment (cmoulliard) [inline openshift/certificate_renewal.adoc:16]: `an OCP cluster certificates.` -> `an OCP cluster certificate.`

File: openshift/certificate_renewal.adoc [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+== References
+
+* https://docs.openshift.com/container-platform/4.10/security/certificates/replacing-default-ingress-certificate.html
+
+== Obtain the Issuer certificate
```
Comment (cmoulliard) [inline openshift/certificate_renewal.adoc:22]: The text to be documented here should be something like:
"By default, the certificate issued during the installation of the OpenShift cluster on RHOS do not include the DNS names defined to access the console, API, oauth, etc services. In this case, it is needed to use a Certificate authority like Letsencrypt to trust a certificate created for different DNS names and using a domain name hosted by a DNS provider (godaddy, cloudfare, etc).

such a process can be managed by the cert-manager when we issue a certificate CR packaging the ACME DNS01 solver and able to contact the DNS provider to validate if the DNS names to be used are valid/exist.

IF this is the case, then letsencrypt will issue a Certificate for the DNS names and will include part of the certificate their CA root certificate R3 

By doing that, the worker nodes will be able to access the API server running on the masters, the users will not get anymore a certificate not valid from their browser if of course you following the instruction detailed here: https://docs.openshift.com/container-platform/4.12/security/certificates/replacing-default-ingress-certificate.html

File: ansible/roles/k8s_cert_manager/README.adoc (hunk 2 above)
Comment (jacobdotcosta) [reply ansible/roles/k8s_cert_manager/README.adoc]: fixed

File: openshift/certificate_renewal.adoc (hunk 4 above)
Comment (jacobdotcosta) [reply openshift/certificate_renewal.adoc:43]: updated the docs.

File: openshift/certificate_renewal.adoc (hunk 3 above)
Comment (jacobdotcosta) [reply openshift/certificate_renewal.adoc]: fixed

File: ansible/roles/k8s_cert_manager/README.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+=== Requirements
+
+TODO
```
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply ansible/roles/k8s_cert_manager/README.adoc]: fixed.

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**